src/cmd/factory/impl/common.go

```diff
@@ -192,29 +193,54 @@ type collection struct {
 
 func buildCollections(
 	service path.ServiceType,
-	tenant, user string,
+	tenant, resource string,
 	restoreCfg control.RestoreConfig,
 	colls []collection,
 ) ([]data.RestoreCollection, error) {
-	collections := make([]data.RestoreCollection, 0, len(colls))
+	var (
+		collections = make([]data.RestoreCollection, 0, len(colls))
+		mc          data.Collection
+		pth         path.Path
+		err         error
+	)
 
 	for _, c := range colls {
-		pth, err := path.Build(
-			tenant,
-			user,
-			service,
-			c.category,
-			false,
-			c.PathElements...)
-		if err != nil {
-			return nil, err
-		}
-
-		mc := exchMock.NewCollection(pth, pth, len(c.items))
-
-		for i := 0; i < len(c.items); i++ {
-			mc.Names[i] = c.items[i].name
-			mc.Data[i] = c.items[i].data
+		if service == path.ExchangeService {
```

Comment:
two requests:
1. replace the if/elseif conditionals for a switch.
2. move the inner code to their own functions.